packages/opentelemetry-exporter-prometheus/src/prometheus.ts

```diff
@@ -117,6 +118,15 @@ export class PrometheusExporter implements MetricExporter {
     this.stopServer(cb);
   }
 
+  /**
+   * Set the pulling callback will be called on request of metrics.
+   *
+   * @param cb The callback
+   */
+  setPullCallback(cb?: () => void) {
```

Comment:
We should specify in the tsdoc that it should be synchronous, otherwise the metrics will be exported while the pull callback is still running.